drgn_tools/task.py

```diff
@@ -203,6 +228,561 @@ def show_tasks_last_runtime(tasks: Iterable[Object]) -> None:
     print_table(rows)
 
 
+def is_task_in_state(task: drgn.Object, state: int) -> bool:
```

Comment:
My immediate thought about this helper is that it would be much more straightforward to have `task_state()` as a helper, to handle the difference in struct member name. Then callers could write their own if-statements such as: `if task_state(tsk) == TASK_RUNNING`.

Having dedicated `is_task_STATE()` functions adds a lot of mostly rote, duplicate code. I do get that it's a bit more ergonomic to be able to just import one function and use it directly, but I'm not sure if the ergonomics justifies the extra code. The same goes for the `for_each_STATE_task()` and `count_STATE_tasks()`. 

I think it would be good to have `task_state()`, `for_each_task_in_state()`, and `count_tasks_in_state()`, and ditch the rest of them. But I'm open to discussion on it!